pkg/operation/botanist/health_check.go

```diff
@@ -507,18 +507,18 @@ func (b *Botanist) checkControlPlane(
 ) (*gardencorev1beta1.Condition, error) {
 	cluster, err := gardenerextensions.GetCluster(ctx, b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
 	if err != nil {
-		msg := "failed to start control plane health check"
 		if apierrors.IsNotFound(err) {
-			msg = "control plane is not yet ready"
+			c := checker.FailedCondition(condition, "ControlPlaneNotReady", "Control plane is not yet ready because of missing cluster resource.")
+			return &c, nil
 		}
-		b.Logger.Errorf("%s: %v", msg, err)
-		return nil, errors.New(msg)
+		b.Logger.Errorf("Failed to execute control plane health checks: %v", err)
+		return nil, err
```

Comment:
Shouldn't we also report a failed condition here? Now, the error is only visible in the logs, or? same below

Reply:
Good question! 
My thought was that this is rather an "unspecified" runtime error. The error message will be propagated and reported as `UNKOWN` in the `ControlPlaneHealthy` condition (see [here](https://github.com/gardener/gardener/blob/e34a58af4a6d4edbfb0ec017901559405ee1a7d7/pkg/operation/botanist/health_check.go#L737)). Before this PR, the same was done in case we didn't find the cluster resource which wasn't nice because at least for new shoots, we expect that the cluster resource is missing.

Reply:
OK, sounds good, thanks